DatadogWebViewTracking/Tests/MessageEmitterTests.swift

```diff
@@ -91,7 +91,37 @@ class MessageEmitterTests: XCTestCase {
         // Then
         let message = try XCTUnwrap(receiverMock.messages.firstWebViewMessage)
         guard case let .rum(event) = message else {
-            return XCTFail("not a log message")
+            return XCTFail("not a rum message")
+        }
+
+        let json = JSONObjectMatcher(object: event)
+        XCTAssertEqual(try json.value("attribute1"), 123)
+        XCTAssertEqual(try json.value("attribute2"), "foo")
+        XCTAssertEqual(try json.array("attribute3").values(), ["foo", "bar", "bizz"])
+    }
+
+    func testWhenReceivingTelemetryEvent_itForwardsToTelemetry() throws {
+        // Given
+        let receiverMock = FeatureMessageReceiverMock()
+        let core = PassthroughCoreMock(messageReceiver: receiverMock)
+        let emitter = MessageEmitter(logsSampler: .mockRandom(), core: core)
```

Comment:
cleanup/ To achieve mid-term goals (no managing `core` reference), could we refactor `MessageEmitter` to depend on feature scope instead? It looks trivial and it would clean up the implementation and tests.
```diff
-        let receiverMock = FeatureMessageReceiverMock()
-        let core = PassthroughCoreMock(messageReceiver: receiverMock)
-        let emitter = MessageEmitter(logsSampler: .mockRandom(), core: core)
+        let featureScope = FeatureScopeMock() 
+        let emitter = MessageEmitter(logsSampler: .mockRandom(), featureScope: featureScope)

-        receiverMock.messages.firstWebViewMessage
+        featureScope.messagesSent().firstWebViewMessage
```
The idea is: `MessageEmitter` doesn't need to depend on the entire `core` in order to send messages. Same applies to tests - we don't need to mock the entire `core` in order to test if emitter sends expected message.

Reply:
WebViewTracking does not register any feature, it is enabled per webview instance and only sends messages on the bus. It doesn't need the full scope machinery, with data-store and such.. I start to think that we would need a bit a interface segregation instead of having everything in a feature scope 🤔 

Lets have a further discussion on that and revisit with another PR, how does it sound?

Reply:
I see, indeed. Well, we have [`MessageSending`](https://github.com/DataDog/dd-sdk-ios/blob/723dc0d61198fad2c627fc2579d24bf991b015ef/DatadogInternal/Sources/DatadogCoreProtocol.swift#L57) in place - if mocked, it can make these tests lean.